### DISCUSSION TITLE
Sampling from Behavior subject sequence over specific intervals synced to stimulus presentation

### DISCUSSION THREAD

**Discussion by msarvestani** [bonsai-rx/bonsai#1304] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/12864873/227744200-8cb4f2d2-8a92-42bc-8f0c-14be60bffa79.png)

![image](https://user-images.githubusercontent.com/12864873/227743978-e1284f4e-1b85-4db8-ae3d-dc05cb7fe28a.png)

Hi,
I'm presenting some stimuli in a series (inside 'Flashing Stim'). First, a red dot appears at the center of the screen, then a white dot at random location, then there is a delay that serves as inter-stimulus-interval. I have the mouse position in the behavior subject 'Mouse Position' and I can't figure out how to sample from it during the interval after the white dot and end of the Delay (two arrows in image) in a way that leaves the rest of my workflow functioning properly.

Thanks.
[debug_stim_mouse.zip](https://github.com/bonsai-rx/bonsai/files/11070299/debug_stim_mouse.zip)

**Comment by RoboDoig** [comment, marked as answer]
Hi @msarvestani - do you need the output of flashing stim to produce all the TK vectors?

If you just need to subscribe to MousePosition during the white dot period you could try something like this:

![image](https://user-images.githubusercontent.com/11314305/228526809-c200a2c1-46b6-4eb0-b4e0-2f833356029a.png)

**Reply by msarvestani** [reply]
Thanks, this worked!
